microsoft-edge/webview2/concepts/working-with-local-content.md

```diff
@@ -282,14 +396,49 @@ webView->Navigate(L"https://demo/index.html");
 
 
 <!-- ====================================================================== -->
-## Handling the WebResourceRequested event
+## Loading local content by handling the WebResourceRequested event
 
 Another way you can host local content in a WebView2 control is by relying on the `WebResourceRequested` event.  This event is triggered when the control attempts to load a resource.  You can use this event to intercept the request and provide the local content, as described in [Custom management of network requests](../how-to/webresourcerequested.md).
 
 `WebResourceRequested` allows you to customize the behavior of local content on a per-request basis. This means you can decide which requests to intercept and provide your own content for, and which requests to let the WebView2 control handle normally.  However, customizing the behavior requires more code, such as virtual host mapping, and requires knowledge of HTTP, to be able to construct a proper response. 
 
 From WebView2's perspective, the resource will have come via the network, and WebView2 will adhere to the headers that are set by the app as part of the response. Using the `WebResourceRequested` event is also slower than other approaches, due to the needed cross-process communication and processing per each request.
 
+
+
+<!-- ------------------------------ -->
+#### Considerations for loading local content by handling the WebResourceRequested event
+
+
+###### Origin-based DOM APIs
+
+Local content loaded via WebResourceRequested an http or https URL and a corresponding origin. So APIs that require an origin such as `localStorage` or `indexedDB` will work, and the usual ["same-origin" rules](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) apply.
+
+
+###### DOM APIs requiring secure context
+
+Some newer browser features are limited to https URLs and using WebResourceRequested allows you to replace https URL web resource requests with your own local content and so these browser features are available. This means webcam APIs, geolocation APIs, and notification APIs, among others are all available. See [Secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) for more information.
+
+
+###### Dynamic content
+
+When loading local content via `WebResourceRequested` you specify the local content to load in your event handler. Since you are directly providing the local content you can dynamically modify it in any manner.
```

Comment:
```suggestion
When loading local content via `WebResourceRequested`, you specify the local content to load in your event handler. Since you are directly providing the local content, you can dynamically modify it in any manner.
```
commas insert pause in screenreaders